pkg/codegen/go/gen_program.go

```diff
@@ -238,11 +243,70 @@ func (g *generator) genNode(w io.Writer, n hcl2.Node) {
 	}
 }
 
+func (g *generator) lowerResourceOptions(opts *hcl2.ResourceOptions) (*model.Block, []interface{}) {
+	if opts == nil {
+		return nil, nil
+	}
+
+	var block *model.Block
+	var temps []interface{}
+	appendOption := func(name string, value model.Expression, destType model.Type) {
+		if block == nil {
+			block = &model.Block{
+				Type: "options",
+				Body: &model.Body{},
+			}
+		}
+
+		value, valueTemps := g.lowerExpression(value, destType, false)
+		temps = append(temps, valueTemps...)
+
+		block.Body.Items = append(block.Body.Items, &model.Attribute{
+			Tokens: syntax.NewAttributeTokens(name),
+			Name:   name,
+			Value:  value,
+		})
+	}
+
+	if opts.Parent != nil {
+		appendOption("Parent", opts.Parent, model.DynamicType)
+	}
+	if opts.Provider != nil {
+		appendOption("Provider", opts.Provider, model.DynamicType)
+	}
+	if opts.DependsOn != nil {
+		appendOption("DependsOn", opts.DependsOn, model.NewListType(model.DynamicType))
```

Comment:
This dynamic type isn't sufficient to generate correct code. You'll probably need to provide some additional metadata to generate `pulumi.Resource[]{...}`

Reply:
Yep.